R/input-date.R

```diff
@@ -124,19 +125,103 @@ dateInput <- function(inputId, label, value = NULL, min = NULL, max = NULL,
                `data-date-days-of-week-disabled` =
                    jsonlite::toJSON(daysofweekdisabled, null = 'null')
     ),
-    datePickerDependency
+    datePickerDependencies(sassVariables)
   )
 }
 
-datePickerDependency <- htmlDependency(
-  "bootstrap-datepicker", "1.6.4", c(href = "shared/datepicker"),
-  script = "js/bootstrap-datepicker.min.js",
-  stylesheet = "css/bootstrap-datepicker3.min.css",
-  # Need to enable noConflict mode. See #1346.
-  head = "<script>
-(function() {
-  var datepicker = $.fn.datepicker.noConflict();
-  $.fn.bsDatepicker = datepicker;
-})();
-</script>"
-)
+datePickerDependencies <- function(sassVariables = list()) {
+
+  if (useBsTheme() || length(sassVariables) == 0) {
+    scssDir <- system.file(package = "shiny", "www", "shared", "datepicker", "scss")
+    tmpDir <- tempfile("datepicker-scss")
+    dir.create(tmpDir)
+    outFile <- file.path(tmpDir, "custom.css")
+    sassFunc <- if (useBsTheme()) bootstraplib::bootstrap_sass else sass::sass
+    defaults <- list(
+      sassVariables %OR% "",
+      bsThemeDateDefaults()
+    )
+    sassFunc(
```

Comment:
I want to make sure I understand here. The documentation for how `sass()` deals with nested lists isn't totally clear to me, so I tried this out:

```R
sass::sass(
  list(
    list(list(a=1), list(a=2)),
    "x {a: $a; }"
  )
)
#> /* CSS */
#> x {
#>   a: 2;
#> }
```

Does that mean that the `bsThemeDateDefaults()` will override `sassVariables`?

Reply:
Ahhh, yea, good catch...I was originally thinking that `bsThemeDateDefaults()` would include `!default` flags (that's how `sliderInput()` and `selectInput()` will work), but that doesn't actually work for bootstrap-datepicker since its Sass forces you to override already existing Bootstrap Sass variables. Thus, I think we should switch the ordering and explain in the documentation that `sassVariables` shouldn't include `!default` flags